### PR TITLE
Refactoring assert raises in python algorithm tests part 3

### PR DESCRIPTION
**Pull request by jhaigh0**
**Description of work.**

**Part 2 #35297 must be merged first**

Third part of the work to remove the blanket use of `assertRaises` when testing algorithm input validation in favour of `assertRaisesRegex`. This way we can expect a certain exception rather than accepting any possible `RuntimeException` which could be an unrelated bug (there were a few found during in the tests from this pr).

Main issue is #34570

**To test:**

 - All tests should pass (handled by Jenkins)
 - Code review

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/PythonInterface/test/python/plugins/algorithms/SaveYDATest.py [hunk 1]
```diff
@@ -218,8 +218,9 @@ def _create_workspace(self, ws_2D=True, sample=True, xAx=True, yAxSpec=True, yAx
         if not yAxMt and not yAxSpec:
             ws = CreateWorkspace(DataX=self.data_x, DataY=self.data_y, DataE=np.sqrt(self.data_y), NSpec=1, UnitX="DeltaE")
             LoadInstrument(ws, True, InstrumentName="TOFTOF")
-            ConvertSpectrumAxis(InputWorkspace=ws, OutputWorkspace=ws, Target="theta", EMode="Direct")
-            return ws
+            ConvertSpectrumAxis(InputWorkspace=ws, OutputWorkspace="ws2", Target="theta", EMode="Direct")
+            ws2 = mtd["ws2"]
+            return ws2
```
Comment (robertapplin) [inline Framework/PythonInterface/test/python/plugins/algorithms/SaveYDATest.py:223]:
```suggestion
            ouput_ws = ConvertSpectrumAxis(InputWorkspace=ws, Target="theta", EMode="Direct")
            return ouput_ws 
```

**Review by robertapplin — CHANGES_REQUESTED** [review]
Looks good. Just had one small suggestion to avoid using the ADS in one of the tests

File: Framework/PythonInterface/test/python/plugins/algorithms/SaveYDATest.py (hunk 1 above)
Comment (robertapplin) [reply Framework/PythonInterface/test/python/plugins/algorithms/SaveYDATest.py:223]: ConvertSpectrumAxis will return the output workspace if we do not provide a string to the `OutputWorkspace` property

Comment (jhaigh0) [reply Framework/PythonInterface/test/python/plugins/algorithms/SaveYDATest.py:223]: Okay 👍

**Review by robertapplin — APPROVED** [review]

**Review by robertapplin — CHANGES_REQUESTED** [review]

**Review by robertapplin — APPROVED** [review]

**Pull request merged**